app/controllers/devise_token_auth/confirmations_controller.rb

```diff
@@ -10,7 +10,7 @@ def show
         token_hash = BCrypt::Password.create(token)
         expiry     = (Time.now + @resource.token_lifespan).to_i
 
-        if @resource.sign_in_count > 0
+        if defined? @resource.sign_in_count && @resource.sign_in_count > 0
```

Comment:
@zachfeldman, @lynndylanhurley Guys, there is a bug:
The parentheses are required here around the first `@resource.sign_in_count`, i.e.:
```
if defined?(@resource.sign_in_count) && @resource.sign_in_count > 0
```
Otherwise `defined?` is applied to the whole expression and returns string `"expression"`, which results into assigning expiry to 1 second ahead of now even when `sign_in_count` equals 0.

Reply:
There is a simple snippet to show the difference:
```
irb(main):001:0> defined? some_var
=> nil
irb(main):002:0> defined? some_var && some_var > 0
=> "expression"
irb(main):003:0> defined?(some_var) && some_var > 0
=> nil
```

Reply:
made a PR #1067 fixing this, thank you for report